packages/components/src/FilterBar/_docs/FilterBar.mdx

```diff
@@ -71,19 +71,65 @@ return (
 
 The `filters` prop is an array of attributes used to create the list of filters.
 
-#### Filter attributes
+#### Usage
+
+We also provide a `Filters<ValuesMap>` type you can use for improved type safety.
 
-Each filter has the following attributes:
 ```tsx
-id: string // A unique id for the filter
-name: string // The name used to label the filter, shown in the trigger button
-Component: JSX.Element // A Filter component or a custom component consuming the FilterBarContext
-isRemovable?: boolean
+import { FilterBar, Filters, DateRange, FilterMultiSelect } from "@kaizen/components"
+
+type ValuesMap = {
+  filterId1: string
+  filterId2: string[]
+  filterId3: DateRange
+}
+
+const filters = [
+  {
+    id: "filterId1",
+    name: "Filter 1",
+    Component: <FilterBar.Select {...props} />,
+  },
+  {
+    id: "filterId2",
+    name: "Filter 2",
+    Component: (
+      <FilterBar.MultiSelect {...props}>
+        {/* Compose your structure using subcomponents of FilterMultiSelect */}
+        {() => <FilterMultiSelect.ListBox {...props} />}
+      </FilterBar.MultiSelect>
+    ),
+  },
+  {
+    id: "filterId3",
+    name: "Filter 3",
+    Component: <FilterBar.DateRangePicker {...props} />,
+    isRemovable: true,
+  },
+] satisfies Filters<ValuesMap>
 ```
 
-##### `Component`
+#### Filter attributes
+
+Each filter has the following attributes:
+
+- [`id: string`](#id-string)
+- [`name: string`](#name-string)
+- [`Component: JSX.Element`](#component-jsxelement)
+- [`isRemovable?: boolean`](#isremovable-boolean)
+- [`isUsableWhen?: (state: FiltersState<ValuesMap>) => boolean`](#isusablewhen-state-filtersstatevaluesmap--boolean)
```

Comment:
To make it easier for consumers to understand what to navigate to these for, mainly around the `isUsableWhen`, I think you need to add the description of the feature to these instead of the typescript defs. Cause I'm not sure it's clear enough why someone would visit the `isUsableWhen` section if it doesn't tell them what the attribute solves.

The top two probably could just have what they're used for and not a section of their own, but maybe play around with the idea.